File: data/code_pairs/cpp/694.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <complex>
#include <cmath>
#include <omp.h>

using namespace std;

void initialData(int Nx, int Ny, const vector<double>& x, const vector<double>& y,
                 vector<vector<complex<double>>>& u, vector<vector<complex<double>>>& uold) {
    u.resize(Nx, vector<complex<double>>(Ny));
    uold.resize(Nx, vector<complex<double>>(Ny));

    #pragma omp parallel for collapse(2)
    for (int j = 0; j < Ny; ++j) {
        for (int i = 0; i < Nx; ++i) {
            u[i][j] = 0.5 * exp(-1.0 * (x[i]*x[i] + y[j]*y[j])) * sin(10.0 * x[i] + 12.0 * y[j]);
            uold[i][j] = u[i][j]; // Copying the initialized values from u to uold
        }
    }
}

File: data/code_pairs/fortran/694.f90
MODULE initialdata_mod
    IMPLICIT NONE
CONTAINS
    SUBROUTINE initialdata(Nx, Ny, x, y, u, uold)
        INTEGER, INTENT(IN) :: Nx, Ny
        REAL(KIND=8), DIMENSION(Nx), INTENT(IN) :: x
        REAL(KIND=8), DIMENSION(Ny), INTENT(IN) :: y
        COMPLEX(KIND=8), DIMENSION(Nx, Ny), INTENT(OUT) :: u, uold
        INTEGER :: i, j

        DO j = 1, Ny
            DO i = 1, Nx
                u(i,j) = 0.5d0 * EXP(-1.0d0 * (x(i)**2 + y(j)**2)) * SIN(10.0d0 * x(i) + 12.0d0 * y(j))
                uold(i,j) = u(i,j)
            END DO
        END DO
    END SUBROUTINE initialdata
END MODULE initialdata_mod